stl/inc/array

```diff
@@ -617,9 +617,14 @@ public:
     _DEPRECATE_TR1_NAMESPACE void assign(const _Ty&) {}
 #endif // _HAS_TR1_NAMESPACE
 
-    _CONSTEXPR20 void fill(const _Ty&) {}
+    _CONSTEXPR20 void fill(const _Ty&) {
+        static_assert(is_copy_assignable_v<_Ty>, "cannot fill zero-sized array of non-copy-assignable type");
+    }
 
-    _CONSTEXPR20 void swap(array&) noexcept {}
+    _CONSTEXPR20 void swap(array&) noexcept {
+        static_assert(
+            !is_const_v<_Ty> || _Is_swappable<_Ty>::value, "cannot swap zero-sized array of non-swappable type");

```

Comment:
`!is_const_v` here is non-conforming to my reading of the Standard. [array.members]/4 says that member `swap` has effects equivalent to `swap_ranges(begin(), end(), y.begin())` (where `y` names the function parameter). There are plenty of non-`const` types - `std::mutex`, for example - that we could instantiate `array` with for which that `swap_ranges` call is ill-formed.